### ISSUE TITLE
timeval options do not have symmetric set/get 

### ISSUE THREAD

**Issue by encukou** [python-ldap/python-ldap#95]
Issue description:

Infinite timeout is set using a negative number, but turns into `None` when retreived.
Setting timeout to `None` leads to an internal error.

```pycon
>>> import ldap
>>> conn = ldap.initialize('https://example.com')
>>> conn.set_option(ldap.OPT_NETWORK_TIMEOUT, -1)
>>> print(conn.get_option(ldap.OPT_NETWORK_TIMEOUT))
None
>>> conn.set_option(ldap.OPT_NETWORK_TIMEOUT, None)
Traceback (most recent call last):
  ...
SystemError: <built-in method set_option of LDAP object at 0x7f943715dd78> returned a result with an error set
```

Operating system: Fedora
Python version: 3.6.3
python-ldap version: 3.0.0b1
Imported from https://sourceforge.net/p/python-ldap/bugs/48/

**Comment by tiran** [comment]
Since https://github.com/python-ldap/python-ldap/pull/90/commits/dcb5c0052b5f83508c1c406d8ad0b23e4f5a3e6a it is possible to set timeout to -1 to indicate infinity. In C I can also map None to -1. It's just a couple of lines.

**Comment by tiran** [comment]
``SystemError: <built-in method set_option of LDAP object at 0x7f943715dd78> returned a result with an error set`` is caused by issue #100 and fixed by #101. I'll implement ``None -> -1`` in another PR.
